source/simulator/checkpoint_restart.cc

```diff
@@ -81,8 +81,23 @@ namespace aspect
     void save_critical_parameters (const Parameters<dim> &parameters,
                                    aspect::oarchive &oa)
     {
+      oa << parameters.convert_to_years;
+      oa << parameters.output_directory;
```

Comment:
I disagree with including this parameter. I can imagine plenty of applications for changing this parameter during restarts (e.g. parameter variations from a common checkpoint) and it is really not critical, as long as all files are in the new directory I am always able to restart from there.